kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/internal/WatchConnectionManager.java

```diff
@@ -193,6 +194,13 @@ public void onFailure(IOException e, Response response) {
           return;
         }
 
+        if (response != null && response.code() == HTTP_OK) {
```

Comment:
I would group this block with the rest of the `if then else` that take place when `response != null`. 

Definitely not a blocker though!